components/ContractViewer/ContractTreeNode.tsx

```diff
@@ -0,0 +1,200 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
+/* eslint-disable react-hooks/rules-of-hooks */
+/* eslint-disable react/jsx-key */
+/* eslint-disable jsx-a11y/accessible-emoji */
+import { useState } from 'react'
+
+import Button from '@mui/material/Button'
+import TextField from '@mui/material/TextField'
+import { TreeItem } from '@mui/x-tree-view/TreeItem'
+
+import { state } from './ContractState'
+
+// import { state } from './ContractState'
+
+type ContractTreeNodeProps = {
+  node: any
+  root: any
+  onSelect: (e) => void
+}
+
+const NodeTypeMap = {
+  Deployment: {
+    label: (node) => {
+      let text = node.info.codeAddress
+      if (node.info.codeAddress != node.info.contextAddress) {
+        text += ' @ ' + node.info.contextAddress
+      }
+
+      return (
+        <div className="whitespace-nowrap">
+          <button
+            onClick={() => {
+              state.removeContract(node.info.codeAddress)
+            }}
+          >
+            XX
+          </button>
+          <span>🗂️ {node.info.etherscanInfo.ContractName}</span>
+          <p className="text-xs">{text}</p>
+        </div>
+      )
+    },
+    // widget: (node) => {
+    //   if (node.impls.length > 0) {
+    //     return (node.impls.map(tree => <ContractTreeNode node={tree} />))
+    //   }
+    // }
+  },
+  ContractDefinition: {
+    label: (node) => {
+      return '📜 ' + node.kind + ' ' + node.name
+    },
+  },
+  FunctionDefinition: {
+    label: (node) => {
+      if (!node.name) {
+        if (node.isConstructor) {
+          return <i>🔧🔧 constructor</i>
+        }
+
+        if (node.isFallback) {
+          return <i>🔧🔧 fallback</i>
+        }
+
+        if (node.isReceiveEther) {
+          return <i>🔧🔧 receive</i>
+        }
+
+        return '*unknown function*'
+      }
+
+      return '🔧 function ' + node.name
+    },
+    widget: (node, root) => {
+      if (node.isConstructor) {
+        return null
+      }
+
+      // TODO: support
```

Comment:
In a month from now you and no one else will be able to tell what this comment is about. Put a bit more info so you can have context when you restart working on it.